android/engine/src/test/java/org/smartregister/fhircore/engine/configuration/ConfigurationRegistryTest.kt

```diff
@@ -152,6 +152,7 @@ class ConfigurationRegistryTest : RobolectricTest() {
       "{\"appId\": \"${appId}\", \"id\": \"${id}\", \"fhirResource\": {\"baseResource\": { \"resource\": \"Patient\"}}}"
     val registerConfig =
       configRegistry.retrieveConfiguration<RegisterConfiguration>(ConfigType.Register, configId)
+    Assert.assertTrue(configRegistry.configCacheMap.containsKey(configId))
```

Comment:
I think we need to add a new test case here. We should be testing two cases : one where the cache exists and one where the cache does not.

Reply:
Yea agreed, we need to test the cache miss path